teuthology/task/install/deb.py

```diff
@@ -10,6 +10,21 @@
 
 log = logging.getLogger(__name__)
 
+def _retry_if_eagain_in_output(remote, args):
+    # wait at most 5 minutes
+    with safe_while(sleep=10, tries=30) as proceed:
+        while proceed():
+            stdout = StringIO()
+            try:
+                return remote.run(args=args, stdout=stdout)
+            except run.CommandFailedError:
+                if "Resource temporarily unavailable" in stdout.getvalue().lower():
+                    continue
```

Comment:
i am not sure if this would work. as test nodes are always locked for exclusive use. would be better if we can figure out the other user holding the frontend lock of dpkg. 

i think the first step is to catch `CommandFailedError` exception and run
```sh
sudo fuser -v /var/lib/dpkg/lock-frontend
```
if "Could not get lock" is in the error message, to understand the root cause.

also in newer versions of `apt-get` the error message might vary, see https://github.com/Debian/apt/blob/289ee74dd23cba7e08b08c6c3602bcf4bf8167bc/apt-pkg/contrib/fileutl.cc#L310-L313

so i'd suggest search "Could not get lock" instead.

and you might want to capture the stderr instead of stdout for the error message.

Reply:
Makes sense and fixed it.

Reply:
> i am not sure if this would work. as test nodes are always locked for exclusive use. would be better if we can figure out the other user holding the frontend lock of dpkg.

i still think we need to root cause the issue before using "retry" as the cure. it's like,

> well, have you tried turning it if off and on again?

Reply:
Checked the teuthology logs again more carefully and I couldn't find any clue about who is holding the lock. We need this patch to reproduce it to list all the users of `lock-frontend` to dig it furture.